pkg/alertmanager/alertmanager_test.go

```diff
@@ -373,8 +374,13 @@ func TestSilenceLimits(t *testing.T) {
 	require.EqualError(t, err, "exceeded maximum number of silences: 1 (limit: 1)")
 	require.Equal(t, "", id2)
 
-	// Expire sil1. This should allow sil2 to be inserted.
+	// Expire sil1 and run the GC. This should allow sil2 to be
```

Comment:
From the comment it looks like you run the GC on `sil1` only, but you actually run it on both `sil1` and `sil2`.

Reply:
`sil2` hasn't been set at this time, so the GC just affects `sil1`. `sil2` is no inserted until Line 384.